spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -188,7 +200,12 @@ private void popMessageAndSend() {
 
 		byte[] value = null;
 		try {
-			value = this.boundListOperations.rightPop(this.receiveTimeout, TimeUnit.MILLISECONDS);
+			if (readFromLeft) {
+				value = this.boundListOperations.leftPop(this.receiveTimeout, TimeUnit.MILLISECONDS);
+			}
+			else {
+				value = this.boundListOperations.rightPop(this.receiveTimeout, TimeUnit.MILLISECONDS);
+			}
```

Comment:
Guys, pay attention to this `if...else`.
I think name the option like `rightPop = true/false` (`right-pop` as an XML attr) will make everything clear.
Default to `true` to achiever the current behavior.

Let's collect "+1" and go ahead with other stuff in this PR!